IMSCovidTracker/Views/HomePage.xaml

```diff
@@ -38,62 +41,69 @@
-                                        IsVisible="{Binding Source={x:Reference HomePageView}, Path=BindingContext.IsBusy, Converter={StaticResource BoolInvert}}"
-                                        ItemsSource="{Binding Source={x:Reference HomePageView}, Path=BindingContext.CountryWidgets}">
+                        <CollectionView Grid.Row="3"
+                                        BackgroundColor="Transparent" 
+                                        x:Name="WidgetCollection"
+                                        x:FieldModifier="public"
+                                        VerticalScrollBarVisibility="Never"
+                                        IsVisible="{Binding IsBusy, Converter={StaticResource BoolInvert}}"
+                                        ItemsSource="{Binding CountryWidgets}"
+                                        ItemSizingStrategy="MeasureFirstItem">
                             <CollectionView.ItemsLayout>
-                                <GridItemsLayout Orientation="Vertical" HorizontalItemSpacing="10" VerticalItemSpacing="10" Span="2"/>
+                                <GridItemsLayout Orientation="Vertical" Span="2"/>
                             </CollectionView.ItemsLayout>
                             <CollectionView.ItemTemplate>
                                 <DataTemplate>
-                                    <SwipeView BackgroundColor="Transparent">
+                                    <ContentView>
+                                    <SwipeView Margin="5">
                                         <SwipeView.LeftItems>
                                             <SwipeItems Mode="Execute">
                                                 <SwipeItem Text="Delete"
-                                                            Command="{Binding Source={x:Reference HomePageView}, Path=BindingContext.DeleteWidgetCommand}"
+                                                            Command="{Binding Source={x:Reference WidgetCollection}, Path=BindingContext.DeleteWidgetCommand}"
                                                             CommandParameter="{Binding .}"
                                                             BackgroundColor="Red"/>
                                             </SwipeItems>
                                         </SwipeView.LeftItems>
                                         <SwipeView.RightItems>
                                             <SwipeItems Mode="Execute">
                                                 <SwipeItem Text="View" 
-                                                            Command="{Binding Source={x:Reference HomePageView}, Path=BindingContext.ViewWidgetCommand}" 
+                                                            Command="{Binding Source={x:Reference WidgetCollection}, Path=BindingContext.ViewWidgetCommand}" 
```

Comment:
Does referencing the WidgetCollection and calling the Command really work or is it a typo ?

Reply:
This works, we are accessing the binding context of the collection view which is the binding context of the page